### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by Mordil**

### CODE REVIEW

File: Sources/Currency/Money.swift [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+  /// - Note: This is likely to be more precise than necessary, so it is recommended to prefer use of `amount` which uses "bankers" rounding.
+  var rawAmount: Decimal { get }
+
+  /// The "bankers" rounded amount of money being represented.
+  ///
+  /// The `rawAmount` will be rounded to the significant digits as defined by the `currency.minorUnits`.
+  ///
+  /// See `CurrencyMetadata.minorUnits` and `Foundation.Decimal.RoundingMode.bankers`.
+  /// - Note: This is usually the desired value to work with, as it is as precise as usually needed.
+  var amount: Decimal { get }
```
Comment (thonydam) [inline Sources/Currency/Money.swift]: `rawAmount` is not super meaningful. What we call `Amount` here is not the actual amount but the rounded value.

To be more explicit, should we have:
`rawAmount` => `amount`
`amount`=> `roundedAmount` or `rounded`

File: Sources/Currency/Money.swift [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+    return lhs.amount == rhs.amount
+  }
+  
+  /// Checks if the current `amount` is equivalent  to the provided decimal, after rounding.
+  ///
+  /// As floating point type precisions can vary, doing exact comparisons to raw `amount` values has a high degree of false negatives.
+  /// To get around this, the provided `other` amount will be rounded to the same precision as the `minorUnits` of the Money's currency using the "banker" mode.
+  ///
+  /// See `Money.amount`.
+  /// - Parameter other: The other amount to "bankers' round, then compare against.
```
Comment (thonydam) [inline Sources/Currency/Money.swift]: The parameter description kind of implies that we should pass `other` as the rounded amount. Where we should actually pass the actual `amount`.

**Review by thonydam — APPROVED** [review]
Looks good to me 👍 !

Have minor comments about naming that you can ignore if you feel strongly about it.

File: Sources/Currency/Money.swift (hunk 1 above)
Comment (Mordil) [reply Sources/Currency/Money.swift]: I actually had that original naming, but towards the end I moved to the current naming - as my concern is many people would use `amount` when they really want `roundedAmount`. 99% of the time, people want the every day banker's rounded value.

How do you feel about:
`amount` => `roundedAmount`
`rawAmount` => `exactAmount`

File: Sources/Currency/Money.swift (hunk 2 above)
Comment (Mordil) [reply Sources/Currency/Money.swift]: That's a reasonable misunderstanding - I'll clarify this section.

File: Sources/Currency/Money.swift (hunk 1 above)
Comment (thonydam) [reply Sources/Currency/Money.swift]: 👍

**Pull request merged**